enterprise/api/v1/views/enterprise_group.py

```diff
@@ -46,22 +46,32 @@ def remove_group_membership_records(group, catalog_uuid=None, user_emails=None):
         records_to_delete = records_to_delete.filter((ecu_in_q | pecu_in_q))
 
     records_to_delete_uuids = [record.uuid for record in records_to_delete]
-    records_to_delete.delete()
-    for records_to_delete_uuids_batch in utils.batch(records_to_delete_uuids, batch_size=200):
-        send_group_membership_removal_notification.delay(
-            group.enterprise_customer.uuid,
-            records_to_delete_uuids_batch,
-            catalog_uuid
+    try:
+        records_to_delete.delete()
+        for records_to_delete_uuids_batch in utils.batch(records_to_delete_uuids, batch_size=200):
+            send_group_membership_removal_notification.delay(
+                group.enterprise_customer.uuid,
+                records_to_delete_uuids_batch,
+                catalog_uuid
+            )
+        # Woohoo! Records removed! Now to update the soft deleted records
+        deleted_records = models.EnterpriseGroupMembership.all_objects.filter(
+            uuid__in=records_to_delete_uuids,
         )
-    # Woohoo! Records removed! Now to update the soft deleted records
-    deleted_records = models.EnterpriseGroupMembership.all_objects.filter(
-        uuid__in=records_to_delete_uuids,
-    )
-    deleted_records.update(
-        status=constants.GROUP_MEMBERSHIP_REMOVED_STATUS,
-        removed_at=localized_utcnow()
-    )
-    return len(deleted_records)
+        deleted_records.update(
+            status=constants.GROUP_MEMBERSHIP_REMOVED_STATUS,
+            removed_at=localized_utcnow()
+        )
+        return len(deleted_records)
+    except Exception as exc:
+        failed_deleted_records = models.EnterpriseGroupMembership.all_objects.filter(
+            uuid__in=records_to_delete_uuids,
+        )
+        failed_deleted_records.update(
```

Comment:
So the way we have it now, if one errors, the whole batch all gets set to the same errored time?

Reply:
yep. I think that this is the most efficient way because if we decide to loop through each record to delete and also send the braze email, we would be making multiple calls to braze. let me know if you have thoughts on a better approach.